src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -1156,9 +1122,7 @@ public Long getTimestamp() {
             }
 
             @Override public boolean willContinue() {
-                return RunningTasks.get().withRunningTasks(runningTasks -> {
-                    return runningTasks.containsKey(cookie);
-                });
+                return hasStarted();
```

Comment:
I think more comprehensible to just check if `cookie` has been written yet, rather than relying on this map entry having been created.

(Cannot check `label != null` since that would already be true at step start time when running `node('some-label') {…}`.)